src/test_textnode.py

```
@@ -80,7 +80,9 @@ def test_link_no_url(self):
         self.assertIsInstance(html_node, LeafNode)
         self.assertEqual(html_node.tag, "a")
         self.assertEqual(html_node.value, "This is a link node without url")
-        self.assertEqual(html_node.props, {"href": ""})
+        with self.assertRaises(ValueError):
+            html_node.props
```

Ruff (0.14.10)

Found useless expression. Either assign it to a variable or remove it.

(B018)